pkg/apis/v1/nodeclaim_conversion_test.go

```diff
@@ -0,0 +1,539 @@
+			v1NodeClaim.Status.ImageID = "test-image-id"
+			Expect(v1beta1NodeClaim.Status.ImageID).To(Equal(""))
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			Expect(v1NodeClaim.Status.ImageID).To(Equal(v1beta1NodeClaim.Status.ImageID))
+		})
+		It("should convert v1 nodeclaim capacity", func() {
+			v1NodeClaim.Status.Capacity = v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("13432"),
+				v1.ResourceMemory: resource.MustParse("1332G"),
+			}
+			Expect(v1beta1NodeClaim.Status.Capacity).To(BeNil())
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			Expect(v1NodeClaim.Status.Capacity).To(Equal(v1beta1NodeClaim.Status.Capacity))
+		})
+		It("should convert v1 nodeclaim allocatable", func() {
+			v1NodeClaim.Status.Allocatable = v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("13432"),
+				v1.ResourceMemory: resource.MustParse("1332G"),
+			}
+			Expect(v1beta1NodeClaim.Status.Allocatable).To(BeNil())
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			Expect(v1NodeClaim.Status.Allocatable).To(Equal(v1beta1NodeClaim.Status.Allocatable))
+		})
+		It("should convert v1 nodeclaim conditions", func() {
+			v1NodeClaim.Status.Conditions = []status.Condition{
+				{
+					Status: status.ConditionReady,
+					Reason: "test-reason",
+				},
+				{
+					Status: ConditionTypeDrifted,
+					Reason: "test-reason",
+				},
+			}
+			Expect(v1beta1NodeClaim.Status.Conditions).To(BeNil())
+			Expect(v1NodeClaim.ConvertTo(ctx, v1beta1NodeClaim)).To(BeNil())
+			Expect(v1NodeClaim.Status.Conditions).To(Equal(v1beta1NodeClaim.Status.Conditions))
+		})
+	})
+})
+
+var _ = Describe("Convert V1beta1 to V1 NodeClaim API", func() {
+	BeforeEach(func() {
+		v1NodePool = &NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-nodepool",
+			},
+			Spec: NodePoolSpec{
+				Template: NodeClaimTemplate{
+					Spec: NodeClaimSpec{
+						NodeClassRef: &NodeClassReference{
+							Name:  "test",
+							Kind:  "test-kind",
+							Group: "test-group",
+						},
+						Requirements: []NodeSelectorRequirementWithMinValues{},
+					},
+				},
+			},
+		}
+		v1beta1NodePool = test.NodePool()
+		v1NodeClaim = &NodeClaim{}
+		v1NodeClaim.Labels = map[string]string{
+			NodePoolLabelKey: v1NodePool.Name,
+		}
+		v1beta1NodeClaim = &v1beta1.NodeClaim{}
+		v1beta1NodeClaim.Labels = map[string]string{
+			NodePoolLabelKey: v1beta1NodePool.Name,
+		}
+		Expect(env.Client.Create(ctx, v1beta1NodePool)).To(BeNil())
+		cloudProvider.NodeClassGroupVersionKind = []schema.GroupVersionKind{
+			{
+				Group:   "fake-cloudprovider-group",
+				Version: "fake-cloudprovider-version",
+				Kind:    "fake-cloudprovider-kind",
+			},
+		}
+		ctx = injection.WithNodeClasses(ctx, cloudProvider.GetSupportedNodeClasses())
+		ctx = injection.WithClient(ctx, env.Client)
+	})
+
+	Context("MetaData", func() {
+		It("should convert v1beta1 nodeclaim Name", func() {
```

Comment:
I wonder if you check all ObjectMeta at once. There's a lot of components of the ObjectMeta and checking that each one was copied one by one isn't probably worth the time as long as we're just doing a straight ObjectMeta mapping